### PR TITLE
Safety check for filter values

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

File: readinglist/resource.py [hunk 1]
```diff
@@ -130,7 +130,11 @@ def _extract_filters(self, queryparams):
             if param in self.known_fields:
                 filters.append((param, value, '=='))
             if param == '_since':
-                filters.append((self.modified_field, value, '>='))
+                if isinstance(value, six.integer_types + (float,)):
```
Comment (Natim) [inline readinglist/resource.py]: Why would you need to handle float here?

File: readinglist/resource.py [hunk 2]
```diff
@@ -130,7 +130,11 @@ def _extract_filters(self, queryparams):
...
+                    filters.append((self.modified_field, value, '>='))
```
Comment (tarekziade) [inline readinglist/resource.py]: I would have a set of constants for stuff like '>=' and '=='.  I have not checked but maybe Python lib has them already

File: readinglist/resource.py (hunk 1 above)
Comment (leplatrem) [inline readinglist/resource.py]: timestamps were decimal values in the first version of #57

File: readinglist/resource.py (hunk 2 above)
Comment (leplatrem) [inline readinglist/resource.py]: instead we could use ``operator.eq`` directly (or ``ast.Eq``)

File: readinglist/resource.py (hunk 1 above)
Comment (Natim) [inline readinglist/resource.py]: So I think we don't need it anymore, right?

File: readinglist/resource.py (hunk 2 above)
Comment (Natim) [inline readinglist/resource.py]: I found it more explicit without constant don't you?

File: readinglist/resource.py (hunk 1 above)
Comment (leplatrem) [inline readinglist/resource.py]: Yes take it out, we'll restore it in case we change our mind in #57

File: readinglist/resource.py (hunk 2 above)
Comment (leplatrem) [inline readinglist/resource.py]: the only advantage I about using constants is to have an exhaustive list somewhere.
(think backend implementation https://github.com/mozilla-services/readinglist/blob/safety-check-for-filter-values/readinglist/backend/memory.py#L55-L62)

Comment (Natim) [inline readinglist/resource.py]: Oh yeah.

Comment (tarekziade) [inline readinglist/resource.py]: The benefit with constants is avoiding typos in the string. You import and uses variables everywhere in the code.

File: readinglist/utils.py [hunk 3]
```diff
@@ -18,9 +18,19 @@ def native_value(value):
...
     return type('Enum', (), enums)
+
+
+COMPARISON = Enum(
+    LT='<',
+    MIN='>=',
+    MAX='<=',
+    NOT='!=',
+    EQ='==',
+    GT='>',
```
Comment (tarekziade) [inline readinglist/utils.py]: That's pretty cool.   I would also move the operators dict in utils  

```python
OPERATORS = {
  COMPARISON.LT: operator.lt,
  COMPARISON.MAX: operator.le,
  COMPARISON.EQ: operator.eq,
  COMPARISON.NOT: operator.ne,
  COMPARISON.MIN: operator.ge,
  COMPARISON.GT: operator.gt,
}
```

Comment (Natim) [inline readinglist/utils.py]: It is what I did and we finally decided it wouldn't make sense since it is related to the memory backend only.

Comment (Natim) [inline readinglist/utils.py]: See https://github.com/mozilla-services/readinglist/commit/4a43c90dc215f63b8c3d6e33c9e2ce5a3ddd4a73#commitcomment-9439133

Comment (tarekziade) [inline readinglist/utils.py]: oh ok then

**Pull request merged**